src/pages/watchlater/index.js

```diff
@@ -1,10 +1,36 @@
+import { useEffect } from "react";
 import Drawer from "../../components/Drawer";
+import { useVideoList } from "../../context/videoListContext";
+import VideoCard from "../../components/Common/VideoCard";
+import { getWatchLaterVideos } from "../../utils/apiHandler/watchlaterVideoHandler";
+import EmptyBox from "../../components/Common/EmptyPage";
 
 export const Watchlater = () => {
-    return (
-        <div className="video-list-container">
-        <Drawer />
-            THis is Watchlater Page
-        </div>
-    )
-}
\ No newline at end of file
+  const { state, dispatch } = useVideoList();
+  const {watchLaterVideos} = state
+
+  const videoLength = watchLaterVideos.length;
+
+  useEffect(() => {
+    getWatchLaterVideos(dispatch);
+  }, []);
```

Comment:
Instead of running useEffect for once by providing an empty [ ], you can give the `token` as dependency array. So, if the token if available --> then only the useEffect( ) will execute.